SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialViewWorkerConnection.cpp

```diff
@@ -0,0 +1,167 @@
+#include "Interop/Connection/SpatialViewWorkerConnection.h"
+
+#include "SpatialGDKSettings.h"
+#include "SpatialView/CommandRequest.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ConnectionHandler/SpatialOSConnectionHandler.h"
+#include "SpatialView/ViewCoordinator.h"
+
+namespace
+{
+
+SpatialGDK::ComponentData ToComponentData(FWorkerComponentData* Data)
+{
+	return SpatialGDK::ComponentData(SpatialGDK::OwningComponentDataPtr(Data->schema_type), Data->component_id);
+}
+
+SpatialGDK::ComponentUpdate ToComponentUpdate(FWorkerComponentUpdate* Update)
+{
+	return SpatialGDK::ComponentUpdate(SpatialGDK::OwningComponentUpdatePtr(Update->schema_type), Update->component_id);
+}
+
+}  // anonymous namespace
+
+void USpatialViewWorkerConnection::SetConnection(Worker_Connection* WorkerConnectionIn)
+{
+	TUniquePtr<SpatialGDK::SpatialOSConnectionHandler> Handler = MakeUnique<SpatialGDK::SpatialOSConnectionHandler>(WorkerConnectionIn);
+	Coordinator = MakeUnique<SpatialGDK::ViewCoordinator>(MoveTemp(Handler));
+}
+
+void USpatialViewWorkerConnection::FinishDestroy()
+{
+	Coordinator.Reset();
+	Super::FinishDestroy();
+}
+
+void USpatialViewWorkerConnection::DestroyConnection()
+{
+	Coordinator.Reset();
+}
+
+TArray<SpatialGDK::OpList> USpatialViewWorkerConnection::GetOpList()
+{
+	check(Coordinator.IsValid());
+	TArray<SpatialGDK::OpList> OpLists;
+	OpLists.Add(Coordinator->Advance());
+	return OpLists;
+}
+
+Worker_RequestId USpatialViewWorkerConnection::SendReserveEntityIdsRequest(uint32_t NumOfEntities)
+{
+	check(Coordinator.IsValid());
+	return Coordinator->SendReserveEntityIdsRequest(NumOfEntities);
+}
+
+Worker_RequestId USpatialViewWorkerConnection::SendCreateEntityRequest(TArray<FWorkerComponentData> Components, const Worker_EntityId* EntityId)
+{
+	check(Coordinator.IsValid());
+	const TOptional<Worker_EntityId> Id = EntityId ? *EntityId  : TOptional<Worker_EntityId>();
+	TArray<SpatialGDK::ComponentData> Data;
+	Data.Reserve(Components.Num());
+	for (auto& Component : Components)
+	{
+		Data.Emplace(SpatialGDK::OwningComponentDataPtr(Component.schema_type), Component.component_id);
+	}
+	return Coordinator->SendCreateEntityRequest(MoveTemp(Data), Id);
+}
+
+Worker_RequestId USpatialViewWorkerConnection::SendDeleteEntityRequest(Worker_EntityId EntityId)
+{
+	check(Coordinator.IsValid());
+	return Coordinator->SendDeleteEntityRequest(EntityId);
+}
+
+void USpatialViewWorkerConnection::SendAddComponent(Worker_EntityId EntityId, FWorkerComponentData* ComponentData)
```

Comment:
```suggestion
void USpatialViewWorkerConnection::SendAddComponent(Worker_EntityId EntityId, FWorkerComponentData&& ComponentData)
```
maybe, since we take ownership and the pointer must be valid